### DISCUSSION TITLE
关于自定义字段来文件夹格式修改的具体操作方法请教

### DISCUSSION THREAD

**Discussion by Arkyuyuzi** [hect0x7/JMComic-Crawler-Python#314] [Q&A] [answered]
我目前是想自定义一个字段myname，然后可以生成 Bd / Aauthor / Amyname的文件夹格式

dir_rule: # 忽略base_dir配置项
  rule: Bd_Amyname # A表示本子，myname表示本子的一个自定义字段
  from jmcomic import JmModuleConfig
# Pmyname
JmModuleConfig.PFIELD_ADVICE['myname'] = lambda photo: f'【{photo.id}】{photo.title}'

原谅是我实在没有计算机语言基础，看了您写的教程，问了半天ai，得到上面这段代码应该是放到option.py这个文件内，这是豆包给的放置位置
![2f664d0dc534c8e2fb3ff8cf5298be02](https://github.com/user-attachments/assets/73354bf2-b817-4cc8-aa91-d8dbbc242116)

这是chatgpt给的放置办法
![71200fe21204fee73a68779be540dea5](https://github.com/user-attachments/assets/c8b8d928-25c8-466e-af85-060af1b44d96)
两种我都尝试了，也修改了对应的option_workflow_download.yml中的dir_rule 运行结果的压缩包是空白的，没有文件

想问问这个自定义字段具体要放到哪里，哪些地方需要修改，带来麻烦，深感抱歉

**Comment by hect0x7** [comment]
修改option_workflow_download.yml，你是想改GitHub Actions工作流的option文件，对工作流生效吗？

**Comment by hect0x7** [comment]
如果你不是想改工作流，而是有自己的python代码，那么下面是一个最简单的例子：
```python
from jmcomic import *

JmModuleConfig.PFIELD_ADVICE['myname'] = lambda photo: f'【{photo.id}】{photo.title}'

option = create_option_by_str('''
dir_rule:
    rule: Bd / Pmyname
''')
option.download_album(123)
```

**Comment by hect0x7** [comment, marked as answer]
我看了下你的repo，你是要改GitHub Actions啊。GitHub Actions是运行usage/workflow_download.py这个文件，你把这行代码加到文件开头就行了
```python
JmModuleConfig.PFIELD_ADVICE['myname'] = lambda photo: f'【{photo.id}】{photo.title}'
```
也就是下面这样
```python
from jmcomic import *
from jmcomic.cl import JmcomicUI

# 就加这里
JmModuleConfig.PFIELD_ADVICE['myname'] = lambda photo: f'【{photo.id}】{photo.title}'
# 以下是原文件内容
# 下方填入你要下载的本子的id，一行一个，每行的首尾可以有空白字符
jm_albums = '''
...
```

**Comment by hect0x7** [comment]
你下载得到空文件是因为把jmcomic的源码改错了，jmcomic库直接报错了，你还是重新fork一下吧
